react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+// function CustomButton(props) {
+//   return <button>{ props.text }</button>;
+// }
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(prevState => ({
+      isClicked: !prevState.isClicked
+    }));
```

Comment:
Although this is a good use of passing `setState` a function, most of the time it is easier to just pass in an object